packages/moments/src/client/getMomentsDefaultClient.ts

```diff
@@ -0,0 +1,73 @@
+import { MomentsClient } from 'moments';
+import {
+  AuthenticationProvider,
+  AuthenticationProviderHttp,
+  PoapCompass,
+  PoapMomentsApi,
+} from '@poap-xyz/providers';
+
+interface GetMomentsDefaultClientOptions {
+  moments: {
+    momentsBaseUrl?: string;
+    /**
+     * The authentication provider used for authenticating requests to the moments API.
+     * If this is provided, oAuthAuthentication will be ignored.
+     */
+    authenticationProvider?: AuthenticationProvider;
+    /**
+     * The OAuth authentication configuration used for authenticating requests to the moments API.
+     * If this is provided, it will be used to create an `AuthenticationProviderOAuth` instance.
+     */
+    oAuthCredentials?: {
+      clientId: string;
+      clientSecret: string;
+      oAuthServerDomain?: string;
+    };
+  };
+  compass?: {
+    compassApiKey?: string;
+    compassBaseUrl?: string;
+  };
+}
+
+const DEFAULT_OAUTH_SERVER = 'https://accounts.poap.tech';
+
+const getAuthenticationProvider = (
+  params: GetMomentsDefaultClientOptions,
+): AuthenticationProvider => {
+  if (params.moments.authenticationProvider) {
+    return params.moments.authenticationProvider;
+  }
+
+  if (!params.moments.oAuthCredentials) {
+    throw new Error(
```

Comment:
Can we make this a typed error?

Reply:
Is this errors supposed to be catched and do some business depending on it? If not I don't see why it needs to be types. Is an error for the developer to see and fix his/her configuration.

Reply:
Agree with Juan